### DISCUSSION TITLE
TableWidget with InteractsWithPageTable trait not working due to function name conflict

### DISCUSSION THREAD

**Discussion by janwytze** [filamentphp/filament#11554] [Help]
### Package

Widgets

### Package Version

v3.2.32

### How can we help you?

I have a page that implements `ListRecords`, so has a table. To this page I want to add a table widget that interacts with the table on the page. The problem is that the `TableWidget` class requires the `getTablePage(): int` function to be implemented , but the `InteractsWithPageTable` trait requires the `getTablePage(): string` function to be overriden. So two functions with the same name and no args need to be implemented in the widget, which is not possible.

How can I add a tablewidget that can interact with the table on the page?

The code:

```php
class ListPurchases extends ListRecords
{
    use ExposesTableToWidgets;
    
    ...

    protected function getFooterWidgets(): array
    {
        return [
            PurchasesTableConsumptions::class,
        ];
    }
}

class PurchasesTableConsumptions extends TableWidget
{
    use InteractsWithPageTable;

    // Needed by TableWidget base class.
    public function getTablePage(): int
    {
        return 0;
    }

    // Needed by InteractsWithPageTable trait.
    protected function getTablePage(): string
    {
        return ListPurchases::class;
    }
}
```

**Comment by janwytze** [comment]
Well I fixed it by copying `InteractsWithPageTable` and prefixing all fields and some methods with `page`. But I do think this is something that should be fixed. There is no reason to not allow the usage of `InteractsWithPageTable` on a `TableWidget`.

**Reply by nazarii-rabbit** [reply]
Up, also got this error

**Reply by sp4r74cus** [reply]
+1
@janwytze can you please share your copy of `InteractsWithPageTable`? i tried the same but got an error when searching/filtering the main table.
Thanks.

**Comment by dharen008** [comment]
same issue i got. i need tableWidget to be interacted with the page table

**Comment by johan-jnn** [comment]
upping this : it should really be possible to have a table widget interacting with the page's table
